src/aap_eda/api/vault.py

```diff
@@ -0,0 +1,82 @@
+#  Copyright 2024 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import shutil
+import tempfile
+
+import pexpect
+
+
+class AnsibleVaultNotFound(Exception):
+    pass
+
+
+class AnsibleVaultEncryptionFailed(Exception):
+    pass
+
+
+class AnsibleVaultDecryptionFailed(Exception):
+    pass
+
+
+def encrypt_string(password: str, plaintext: str, vault_id: str) -> str:
+    if not shutil.which("ansible-vault"):
```

Comment:
I think we want to raise the error at startup as we do with git: 
https://github.com/ansible/eda-server/blob/main/src/aap_eda/services/project/git.py#L47

The concern of @benthomasson is to not use ansible as library, but even through the CLI, the presence of ansible-vault is still a requirement of the application.